Objects/floatobject.c

```diff
@@ -1208,11 +1201,118 @@ hex_from_char(char c) {
     return x;
 }
 
-/* convert a float to a hexadecimal string */
+/* Convert a float to a hexadecimal string [±][0x]h[.hhhhhhhh]p±d,
+   where the fractional part either is exact (precision < 0) or the
+   number of digits after the dot is equal to the precision.
+
+   The exponent d is written in decimal, it always contains at least one digit,
+   and it gives the power of 2 by which to multiply the coefficient.
+
+   x - the double to be converted
+   precision - the desired precision
+   always_add_sign - nonzero if a '+' sign should be included for x > 0
+   use_alt_formatting - nonzero if the hexadecimal prefix should be added.
+   upper - nonzero, if uppercase letters should be used for hexadecimal
+           numbers, prefix and the exponent separator.
+   _hex - private flag to keep float.hex() behaviour
+ */
+
+char *
+_Py_dg_dtoa_hex(double x, int precision, int always_add_sign,
+                int use_alt_formatting, int upper, int _hex)
+{
+    int e, autoprec = precision < 0;
+    double m = frexp(fabs(x), &e);
+
+    if (autoprec) {
```

Comment:
```suggestion
    int e;
    double m = frexp(fabs(x), &e);
    int autoprec = precision < 0;

    if (autoprec) {
```
Do you mean this?  // per PEP7 "blank line after local variables declaration"

Reply:
I mean grouping e and m, empty line, declare autoprec and useautoproc.

> Do you mean this? // per PEP7 "blank line after local variables declaration"

Python uses C11, we can declare variables in the middle of the function. I prefer to defer variable declaration until it's used (just before its first usage/assignment), to reduce their scope, and so ease review/maintenance (less variables to think about when reviewing).